AI/BattleAI/BattleEvaluator.cpp

```diff
@@ -350,10 +350,11 @@ bool BattleEvaluator::attemptCastingSpell(const CStack * activeStack)
 	LOGL("Casting spells sounds like fun. Let's see...");
 	//Get all spells we can cast
 	std::vector<const CSpell*> possibleSpells;
-	vstd::copy_if(VLC->spellh->objects, std::back_inserter(possibleSpells), [hero, this](const CSpell *s) -> bool
-	{
-		return s->canBeCast(cb->getBattle(battleID).get(), spells::Mode::HERO, hero);
-	});
+
+	for (auto const & s : VLC->spellh->objects)
```

Comment:
Don't see a reason to change this

Reply:
Result of my initial attempt to change `VLC->spellh->objects` to shared_ptr.

Both version are actually not good -  `VLC->spellh->objects` should be a private member eventually, with some clear way to access objects / select matching.

Reply:
@IvanSavenko can it be rewritten to use VLC->spells?